code/modules/hydroponics/grown.dm

```diff
@@ -475,4 +491,21 @@
 			return
 		reagents.remove_any(rand(1,3)) //Todo, make it actually remove the reagents the seed uses.
 		seed.do_thorns(H,src)
-		seed.do_sting(H,src,pick("r_hand","l_hand"))
\ No newline at end of file
+		seed.do_sting(H,src,pick("r_hand","l_hand"))
+
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/On_Consume()
+	if(seed && seed.get_trait(TRAIT_BATTERY_RECHARGE))
+		if(!reagents.total_volume)
+			var/batteries_recharged = 0
+			for(var/obj/item/weapon/stock_parts/cell/C in usr.GetAllContents())
+				var/newcharge = (potency*0.01)*C.maxcharge
+				if(C.charge < newcharge)
```

Comment:
Yeah, it works on eguns, batons, and the like---you can generally get 3 more lethals shots from an energy gun by consuming something with this trait that has a potency of 30 (which is what you're going to be doing, the vast majority of the time).